### PR TITLE
Move EIP Integration to Common

### PR DESCRIPTION
**Pull request by holgerd77**
Follow-up on #856 and #863 

This PR moves EIP integration from the VM to Common to be able to spawn EIP implementations across different libraries touched (as currently possible along hardfork lines). Upon this base integration it updates the EIP2537 integration in the VM.

The Common integration is hopefully somewhat future proof. I already added an additional check for `minimumHardfork` requirements where Common initialization will throw if not met (EIP2537 is currently set to chainstart, let me know if this is too low for some reasons).

Not yet integrated is a possible EIP dependency, e.g. when one EIP depends on one or several others. This could be implemented by adding a parameter `EIPdependencies` or the like to the `EIP*.json` files. Since this comes with some mild extra complexity - e.g. on the param() functionality - I didn't want to integrate directly and eventually overload the PR. Think this can be added once this case occurs.

I also decided to change the semantics of the `param()` function and add changed demarcation lines to the existing `paramByEIP()` and a newly added `paramByHardfork()` function. `param()` removes the `hardfork` parameter and is now the function which just "does the right thing" [tm] - which means it either takes the parameter from the latest hardfork or - if present - from an EIP set. If an EIP is set and contains the parameter, this setting always takes precedence (which I think makes sense and would be expected e.g. on the VM instantiated with a certain HF and an additional EIP changing gas costs e.g.). `paramByEIP()` and `paramByHardfork()` can now be used to explicitly demand for an EIP respectively HF-specific parameter.

This makes use of the`param()` function - especially in the VM - completely transparent and avoids code such as:

```typescript
if (EIP_ACTIVATED) {
  common.paramByEIP(...)
} else {
  common.param()
}
```

Instead the current `param()` usage can stay completely "as is" and the parameter is chosen correctly based on the chain setup in Common.

### CODE REVIEW

**Comment by holgerd77** [comment]
Ah, too many VM API tests still failing, will look into that later.

**Comment by holgerd77** [comment]
Fixed that remaining failing tests.

This is now ready for review. 😄

File: packages/common/src/index.ts [hunk 1]
```diff
@@ -22,6 +22,15 @@ export interface CommonOpts {
    * Limit parameter returns to the given hardforks
    */
   supportedHardforks?: Array<string>
+  /**
+   * Selected EIPs which can be activated, please use an array for instantiation
+   * (e.g. `eips: [ 'EIP2537', ])
```
Comment (ryanio) [inline packages/common/src/index.ts:27]:
```suggestion
   * (e.g. `eips: [ 'EIP2537', ]`)
```

**Review by ryanio — APPROVED** [review]
LGTM, good stuff!

**Pull request merged**